Runtime/JSNativeApi.cs

```diff
@@ -833,45 +731,68 @@ public static void Seal(this JSValue thisValue)
             JSCallback callback = (JSCallback)data!;
             return (napi_value)callback(new JSCallbackArgs(scope, callbackInfo, args));
         }
-        catch (JSException ex)
+        catch (Exception ex)
         {
-            //TODO: [vmoroz] Make JSException to wrap up the original error object instead.
-            ThrowError("", ex.Message);
-            return new napi_value(nint.Zero);
+            JSError.ThrowError(ex);
+            return napi_value.Null;
         }
     }
 
     [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvCdecl) })]
     private static unsafe napi_value InvokeJSMethod(napi_env env, napi_callback_info callbackInfo)
     {
         using var scope = new JSValueScope(JSValueScopeType.Callback, env);
-        JSCallbackArgs.GetDataAndLength(env, callbackInfo, out object? data, out int length);
-        Span<napi_value> args = stackalloc napi_value[length];
-        JSPropertyDescriptor desc = (JSPropertyDescriptor)data!;
-        return (napi_value)desc.Method!.Invoke(
-            new JSCallbackArgs(scope, callbackInfo, args, desc.Data));
+        try
+        {
+            JSCallbackArgs.GetDataAndLength(env, callbackInfo, out object? data, out int length);
+            Span<napi_value> args = stackalloc napi_value[length];
+            JSPropertyDescriptor desc = (JSPropertyDescriptor)data!;
+            return (napi_value)desc.Method!.Invoke(
+                new JSCallbackArgs(scope, callbackInfo, args, desc.Data));
+        }
+        catch (Exception ex)
+        {
+            JSError.ThrowError(ex);
+            return napi_value.Null;
```

Comment:
I think undefined would be more correct than null... though since the error was thrown the caller never gets the result anyway.

Reply:
In Node.JS they expect callbacks to return `NULL` if no data should be returned. The `undefined` will require us to call extra Node-APu function. If we return a non-null value, then the Node.JS code will set result value: https://github.com/nodejs/node/blob/655b070d31efeea2f5580ca221167f8abc43f0e8/src/js_native_api_v8.cc#L323-L325